proto/redpanda/api/console/v1alpha1/pipeline.proto

```diff
@@ -2,22 +2,117 @@ syntax = "proto3";
 
 package redpanda.api.console.v1alpha1;
 
+import "buf/validate/validate.proto";
+import "google/api/field_behavior.proto";
+import "google/protobuf/field_mask.proto";
+import "protoc-gen-openapiv2/options/annotations.proto";
 import "redpanda/api/dataplane/v1alpha2/pipeline.proto";
 
+// RedpandaConnectTier is the tier of the pipeline.
+enum RedpandaConnectTier {
+  REDPANDA_CONNECT_TIER_UNSPECIFIED = 0;
+  // The development tier.
+  REDPANDA_CONNECT_TIER_DEV = 1;
+  // The standard tier.
+  REDPANDA_CONNECT_TIER_STANDARD = 2;
+  // The pro tier.
+  REDPANDA_CONNECT_TIER_PRO = 3;
+  // The custom tier.
+  REDPANDA_CONNECT_TIER_CUSTOM = 4;
```

Comment:
The plan is that only frontend will use tiering, and in the Console version the tier will map to predefined resource requests and limits. The Dataplane API will still use granular requests and limits for resources. since those will not necessarily map to predefined values, we have to categorize them somehow to a tier... so the idea was to use a "custom" enum for that...

Reply:
Okay got you, I think we can make that clear by adding behaviour annotations and make that a read-only flag plus adding validation annotations on the request message so that this option cannot be selected